CMakeLists.txt

```diff
@@ -0,0 +1,76 @@
+cmake_minimum_required(VERSION 3.22)
+
+project(sail_riscv)
+
+# Enable CTest
+enable_testing()
+
+# We technically require C++20 since the C generated by Sail - which we compile
+# as C++ - uses designated initialisers, a C++20 feature. However in practice
+# much older compilers support this feature so everything does work with C++17,
+# but you get lots of warnings on compilers that support C++20 if you use
+# this feature without -std=c++20.
+if (cxx_std_20 IN_LIST CMAKE_CXX_COMPILE_FEATURES)
+    set(CMAKE_CXX_STANDARD 20)
+else()
+    set(CMAKE_CXX_STANDARD 17)
+endif()
+set(CMAKE_CXX_STANDARD_REQUIRED TRUE)
```

Comment:
I think it should also be possible to use the default fallback to older version behaviour:
```
set(CMAKE_CXX_STANDARD 20)
if (cxx_std_20 IN_LIST CMAKE_CXX_COMPILE_FEATURES)
    set(CMAKE_CXX_STANDARD_REQUIRED TRUE)
endif()
```
But as far as I can tell the current sail code does not rely on this and the files are compiled as C?

Reply:
Ah good point - this was needed for the `struct Model { #include "model.c" }` trick which isn't used yet. I'll just remove it for now since there isn't any C++ yet.